docs/docsite/rst/dev_guide/collections_tech_preview.rst

```diff
@@ -384,16 +384,17 @@ This works for roles or any type of plugin distributed within the collection:
 
      - hosts: all
        tasks:
-         - include_role:
-             name : my_namespace.my_collection.role1
```

Comment:
We do not agree that `roles` is less prevalent. If you talk to many of the core developers, we prefer `include/import_role`.

Reply:
Changing this to `import_role` would be preferable since there is no reason for this to be dynamic (`include_role`).

Reply:
So it sounds like it's preferred to use `include_role:`, but not the `roles:` structure? That is the structure that is first mentioned in Ansible's documentation, and is what I see used more often in the wild (especially as include/import_role was not fully fleshed out until recent versions of Ansible).

I know the core team prefers this but I think it would be best to be consistent in documentation and/or move towards changing the other docs to move away from using `roles:` in a playbook... otherwise I think this documentation is more consistent.

Reply:
We don't need or want to move away from using `roles`. But if a role is used in the `tasks` section, the preference is to use `import_role` over `include_role` (since most of the time dynamic role inclusion is not needed).

A playbook consisting only of roles is still better expressed (IMO) using `roles` rather than `tasks` and a bunch of `import_role` tasks. But when you have both `tasks` and `roles` in a play, the execution order is clearer if everything is in `tasks`. This is the main reason `include/import_role` is preferred over `roles`.

In this example, we have a play with `tasks` and `roles` in which case listing everything in order it `tasks` makes for a more intuitive order of events when the play is run.

I would either change this to `import_role` or move the `roles` section above `tasks`. Even though order doesn't matter, it would be nice to have the playbook read in the order in which it will be executed.